api/agent_client_test.go

```diff
@@ -50,15 +50,34 @@ func TestCreateAgent(t *testing.T) {
 		require.Empty(t, status)
 	})
 
-	t.Run("missing ucConfig", func(t *testing.T) {
-		status, err := agent.Create(&loadtest.Config{}, nil)
+	t.Run("missing uc type", func(t *testing.T) {
```

Comment:
Nit: this won't hurt to be here. But this should be taken care with `defaults` package at upper level. @lieut-data have you came across this issue while using it? If so, we should probably do a `defaults.Validate` somewhere.

We define this like:
```golang
Type userControllerType `default:"simulative" validate:"oneof:{simple,simulative,noop,cluster,generative}"`
```

Reply:
I haven't personally come across this no -- I think this is the existing test case that I just reordered to extend coverage of the various `ucConfig` permutations below.

Reply:
Ah, ok. Thanks :)